src/command.coffee

```diff
@@ -441,14 +442,79 @@ parseOptions = ->
 
 # The compile-time options to pass to the CoffeeScript compiler.
 compileOptions = (filename, base) ->
-  answer = {
-    filename
+  if opts.transpile
+    # The user has requested that the CoffeeScript compiler also transpile
+    # via Babel. We use Babel as an `optionalDependency`; see
+    # https://docs.npmjs.com/files/package.json#optionaldependencies.
+    try
+      require 'babel-core'
+    catch
+      console.error '''
+        To use --transpile, you must have Babel installed and configured.
+        See http://coffeescript.org/#usage
+      '''
+      process.exit 1
+
+    # We’re giving Babel only a string, not a filename or path to a file, so
+    # it doesn’t know where to search to find a `.babelrc` file or a `babel`
```

Comment:
Just came across this PR. Could you clarify this? Why can't you pass Babel the filepath and let it resolve the config on its own?

In Babel 7 we're adding more ways to configure Babel, some of which are programmatic with an API managed by `babel-core` so doing stuff like this is going to get harder. Even for config files we're adding caching logic for performance, so by doing this all of that gets sidestepped which will hurt performance. I'd like to understand what limitations exist that made this necessary, so we can fix them if possible.